internal/services/notification/webhooks.go

```diff
@@ -115,3 +125,70 @@ func (n *NotificationService) generatewebhook(ctx context.Context, ev *rstypes.R
 
 	return webhook
 }
+
+func (n *NotificationService) runWebhooksCleanerLoop(ctx context.Context, runWebhookExpireInterval time.Duration) {
+	n.log.Debug().Msgf("webhookCleanerLoop")
+
+	for {
+		if err := n.runWebhooksCleaner(ctx, runWebhookExpireInterval); err != nil {
+			n.log.Warn().Err(err).Msgf("webhooksCleaner error")
+		}
+
+		sleepCh := time.NewTimer(runWebhookExpireInterval).C
+		select {
+		case <-ctx.Done():
+			return
+		case <-sleepCh:
+		}
+	}
+}
+
+func (n *NotificationService) runWebhooksCleaner(ctx context.Context, runWebhookExpireInterval time.Duration) error {
+	l := n.lf.NewLock(webhooksCleanerLockKey)
+	if err := l.TryLock(ctx); err != nil {
+		if errors.Is(err, lock.ErrLocked) {
+			return nil
+		}
+		return errors.WithStack(err)
+	}
+	defer func() { _ = l.Unlock() }()
+
+	for {
+		var runWebhooks []*nstypes.RunWebhook
+
+		err := n.d.Do(ctx, func(tx *sql.Tx) error {
+			var err error
+			runWebhooks, err = n.d.GetRunWebhooks(tx, maxRunWebhooksQueryLimit)
```

Comment:
Looks like you're iterating over the same results forever if the length is greater than maxRunWebhooksQueryLimit. isn't there a missing start id?

And looks like the tests aren't covering this case.